jax/_src/distributed.py

```diff
@@ -34,8 +35,9 @@ def initialize(self,
                  coordinator_address: Optional[str] = None,
                  num_processes: Optional[int] = None,
                  process_id: Optional[int] = None):
-    coordinator_address = os.environ.get('JAX_COORDINATOR_ADDRESS',
-                                         None) or coordinator_address
+    self.process_id = process_id
+    coordinator_address = (coordinator_address or
```

Comment:
Note this change. This is intentional: if someone specifies a specific value for an arg, an environment variable should not override it.